src/core/DY.Core.Trace.Base.fst

```diff
@@ -298,27 +298,69 @@ let rand_generated_at tr i b =
 /// Functions to replace labels in a trace by `()`.
 /// This is useful to use labels that are predicates on `trace_ unit`
 /// with actual traces that are `trace_ label`.
-/// See `DY.Core.Label.Type` for more information
-
-val trace_event_forget_labels:
-  trace_event ->
-  trace_event_ unit
-let trace_event_forget_labels ev =
+/// See `DY.Core.Label.Type` for more information.
+/// This is defined using a `fmap` function on traces,
+/// which can help proving properties on `trace_forget_labels`.
+
+val fmap_trace_event:
```

Comment:
The name `fmap_trace_event` sounds way more general, than what this function actually does. (I.e., for `fmap_trace`, I expected something like applying `f` to every event in the trace)
I think it gets more clear, if we add a "label" somewhere. Maybe `fmap_label_trace_event` or the other way around `label_fmap...`?

Reply:
I took inspiration from [Haskell's `fmap`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Data-Functor.html) because it matches its type signature (with `f` being either `trace_event_` or `trace_`), would it help if I explained more thoroughly in the comment where name comes from and that it only applies the function on labels, or would the name still be confusing?

Reply:
I added comments about that, is it better? I would prefer keeping the name `fmap_trace_event` because it is the `fmap` instantiation on the `trace_event_` functor, so it follows similar conventions to the ones we use e.g. for typeclass instances (although there is no functor typeclass, it is similar to one)

Reply:
The comment definitely helps!!
I also understand the naming, but I still think, it would help, if the name already contained something with "label" (instead of having to look at the comments to get to know, that this is only about labels).
However, this is not blocking the rest of the useful changes in this PR, so we can leave it like this for now.